aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -1018,6 +1018,23 @@ Tensor& heaviside_(Tensor& self, const Tensor& values) {
   return at::heaviside_out(self, self, values);
 }
 
+Tensor& ldexp_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  TORCH_CHECK(!self.is_complex(), "ldexp does not support complex tensors.");
+  TORCH_CHECK(result.dtype() == self.dtype(), "ldexp result dtype must match input dtype.");
```

Comment:
This is no longer true because the composite ops will handle the type promotion for you.